src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/AbstractCatalogBackupRestoreTasklet.java

```diff
@@ -0,0 +1,465 @@
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.util.Assert;
+
+import com.thoughtworks.xstream.XStream;
+
+/**
+ * TODO
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public abstract class AbstractCatalogBackupRestoreTasklet implements Tasklet, InitializingBean {
+
+    /*
+     * 
+     */
+    protected static Map<String, Filter<Resource>> resources = new HashMap<String, Filter<Resource>>();
+
+    /*
+     * 
+     */
+    static {
+        resources.put("demo", AnyFilter.INSTANCE);
+        resources.put("images", AnyFilter.INSTANCE);
+        resources.put("logs", new Filter<Resource>() {
+
+            @Override
+            public boolean accept(Resource res) {
+                if (res.name().endsWith(".properties")) {
+                    return true;
+                }
+                return false;
+            }
+            
+        });
+        resources.put("palettes", AnyFilter.INSTANCE);
+        resources.put("plugIns", AnyFilter.INSTANCE);
+        resources.put("styles", new Filter<Resource>() {
+
+            @Override
+            public boolean accept(Resource res) {
+                if (res.name().endsWith(".sld") || res.name().endsWith(".xml")) {
+                    return false;
+                }
+                return true;
+            }
+            
+        });
+        resources.put("user_projections", AnyFilter.INSTANCE);
+        resources.put("validation", AnyFilter.INSTANCE);
+        resources.put("www", AnyFilter.INSTANCE);
+    }
+    
+    protected Backup backupFacade;
+
+    protected Catalog catalog;
+
+    protected XStreamPersisterFactory xStreamPersisterFactory;
+
+    protected XStreamPersister xstream;
+
+    protected XStream xp;
+    
+    private boolean isNew;
+
+    private AbstractExecutionAdapter currentJobExecution;
+
+    private boolean dryRun;
+
+    private boolean bestEffort;
+
+    public AbstractCatalogBackupRestoreTasklet(Backup backupFacade,
+            XStreamPersisterFactory xStreamPersisterFactory) {
+        this.backupFacade = backupFacade;
+        this.xStreamPersisterFactory = xStreamPersisterFactory;
+    }
+    
+    @Override
+    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) 
+            throws Exception {
+        // Accordingly to the running execution type (Backup or Restore) we
+        // need to validate resources against the official GeoServer Catalog (Backup)
+        // or the temporary one (Restore).
+        //
+        // For restore operations the order matters.
+        JobExecution jobExecution = chunkContext.getStepContext().getStepExecution()
+                .getJobExecution();
+        this.xstream = xStreamPersisterFactory.createXMLPersister();
+        if (backupFacade.getRestoreExecutions() != null
```

Comment:
Same pattern as in other places referring back to the facade, here the execution context is a bit harder to get but it's still feasible, see: http://stackoverflow.com/questions/8117060/storing-in-jobexecutioncontext-from-tasklet-and-accessing-in-another-tasklet